### PR TITLE
Add FieldGroup and Input components

### LINKED ISSUE #558: Create TextInput and TextArea inputs
Parent #550

## Explanation
Team Front-end has provided us with loads of CSS and HTML files that we can use to create React components.

This issue relates to the input components.

## Tasks
- [x] Create a React component from the HTML
- [x] Write tests for the component
- [x] Add the new component to the `Reactification` tab in the example components app
- [x] Update snapshots for the example app
- [x] Make sure `eslint` is OK

### PR DESCRIPTION
**Pull request by bintzandt**
## Summary
This PR adds the new `Input` components to `@yoast/components`.

This PR can be summarized in the following changelog entry:
* [yoast-components] Create new React components based on the HTML that was already provided.

## Relevant technical choices:
* A `FieldGroup` component has been created that can be used to wrap the `Input` fields with a label, description and a `HelpIcon`.
* The `description` of the `FieldGroup` renders conditionally based on the length of the description. If the length is 0, we assume that no description has been provided and we will therefore not render a `<p>`.
* I've created separate files for `TextInput` and `TextArea`. This seemed like the most logical decision at the moment. If it turns out that there is a better way to categorize the components than please go ahead :)

## Tasks
- [x] Add a `TextInput` component
- [x] Add a `TextArea` component
- [x] Write tests for the `TextInput` component
- [x] Write tests for the `TextArea` component

## Test instructions

_Test instructions will be written when the PR is finished._
This PR can be tested by following these steps:
* Check that the Example components app works as expected

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [x] I have added unittests to verify the code works as intended

Parent #550 

Fixes #558

### CODE REVIEW

**Comment by bintzandt** [comment]
Two question that still remain:
1. Do we want to merge this into the feature branch? Or do we want to merge this into develop directly?
1. What needs to happen if we want to change the export of the old `Input` component to the new `TextInput` component?

File: packages/components/src/inputs/TextArea.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Renders a textarea for use in our HTML forms.
+ *
+ * @param {object} props The properties required for rendering this component.
+ *
+ * @returns {React.Component} A react component that can be used in our forms.
+ */
+const TextArea = ( props ) => {
+	const id = getId( props.id );
```
Comment (maartenleenders) [inline packages/components/src/inputs/TextArea.js]: 🤔 I was thinking...
Is it not preferable to have `const id = props.id || null;` (or even have that as the default prop)?

That way `id` will only be set if there is one set explicitly.

If you are worried about similar items, you can always set the `key`.

Comment (maartenleenders) [reply packages/components/src/inputs/TextArea.js]: Ah nvm you need the `id` for the labels, my bad

File: packages/components/src/inputs/TextInput.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			<input
+				id={ id }
+				name={ props.name }
+				defaultValue={ props.value }
+				type={ props.type }
+				className="yoast-field-group__inputfield"
+				aria-describedby={ props.ariaDescribedBy }
+				placeholder={ props.placeholder }
+				readOnly={ props.readOnly }
+				min={ props.min }
```
Comment (maartenleenders) [inline packages/components/src/inputs/TextInput.js]: I don't necessarily like passing irrelevant props to the input object, even if they are undefined (I believe they end up in the DOM as `<input id="bla" min max />`, correct?).

Isn't it better to not strictly set these props in advance, like we have now with the `undefined` defaultProps, but to have the input here saying:

```js
<input
	{ ...props }
/>
```
That way, if we don't add `min` or `max` to our `<TextInput id="bla" type="text" />` call, it won't appear in the DOM. Makes things a bit more readable, I think?

I might be missing something obvious, though?

**Comment by maartenleenders** [comment]
CR: small suggestion, could you have a look and tell me what you think / why you wanted to do it differently?

Regarding your question, bit out of my depth but
1) social preview feature branch doesn't seem logical, we need text inputs elsewhere. A new feature branch also doesn't seem great.
If I am not mistaken, once our plugin can load the CSS, we can just export these new components without using them, so `develop` seems fine **if we don't release monorepo before our plugin is updated with the css-loader**. Let's discuss in the standup.
2) If they look exactly on the implementation side, perhaps not so much? I don't have an overview of how they differ currently.

File: packages/components/src/inputs/TextInput.js (hunk 2 above)
Comment (bintzandt) [reply packages/components/src/inputs/TextInput.js]: The props **do** en up in the virtual DOM that React uses.

However, they are **not** present in the actual HTML DOM.

As to your suggestion: ESLint complains if there are props that are not required and that do not have a default value. Therefore, I have decided to provide a default value of `undefined` to make sure that they do not end up in the real (HTML) DOM.

If you like it better, I could opt for the following:
```jsx
<input
	id={ id }
	defaultValue={ props.value }
	className="yoast-field-group__inputfield"
	aria-describedby={ props.ariaDescribedBy }
	{ ...props }
/>
```

**Comment by maartenleenders** [comment]
Acceptance 👍 

**BUT:** 
The `input.html` looks a bit different from the one in the example app. Not talking about the backgrounds so much, but about the borders. I think this is because the html loads the `borders.css` stylesheet (which loads some border styles into the root style), whereas we only import `input.css`. 

We should probably have a default css import helper somewhere, that will import all these related things. Do you agree?
 Holding the merge until we discuss 😉

**Comment by maartenleenders** [comment]
Base styles issue is non-blocking, will be implemented later.

**Pull request merged**